### PR TITLE
Color palette refactor

### PR DESCRIPTION
**Pull request by eligijusTrafi**
cc @mikasdo @justasm

### CODE REVIEW

**Comment by mikasdo** [comment]
Need to look at how we expose GrayScale customisation for SDK users

File: android/ui/src/main/java/com/trafi/ui/theme/Color.kt [hunk 1]
```diff
@@ -117,6 +113,8 @@ class MaasColorPalette(
         private set
     var onError by mutableStateOf(onError)
         private set
+    var grayScale by mutableStateOf(grayScale)
```
Comment (justasm) [inline android/ui/src/main/java/com/trafi/ui/theme/Color.kt:116]: What do you think about calling this `gray` and the various variants just `100`, `200`, etc? So we could write `MaasTheme.colors.gray.100` 🤔

File: common/core/src/commonMain/kotlin/com/trafi/ui/theme/internal/Color.kt [hunk 2]
```diff
@@ -11,4 +11,20 @@ import kotlin.jvm.JvmName
 
 expect inline class Color(val value: ULong)
 
+internal fun Color.alpha(value : Float): Color {
```
Comment (justasm) [inline common/core/src/commonMain/kotlin/com/trafi/ui/theme/internal/Color.kt:14]: Very clear Kotlin function, super nice job @mikasdo 👏 Good visibility modifier, using `require` to verify inputs, clear variable names 💪

**Review by justasm — COMMENTED** [review]
Looking good! Do you plan to change the existing palette to add the "warning" and "success" colors defined in Figma?

File: android/ui/src/main/java/com/trafi/ui/theme/Color.kt (hunk 1 above)
Comment (eligijusTrafi) [reply android/ui/src/main/java/com/trafi/ui/theme/Color.kt:116]: 🔝 good idea 🤔

Comment (justasm) [reply android/ui/src/main/java/com/trafi/ui/theme/Color.kt:116]: 😮 what if wrapped in backticks, e.g.
```
val `100` = ..
```
Not pretty though, so probably not the way to go

**Comment by eligijusTrafi** [comment]
Note that `Warning` , `Success` and `Danger` colors cannot be changed by outsiders right now.

File: android/ui/src/main/java/com/trafi/ui/theme/Color.kt [hunk 3]
```diff
@@ -8,16 +8,7 @@ import androidx.compose.runtime.mutableStateOf
...
-@Stable val Grey100 = ColorPalette.Gray100
-@Stable val Grey200 = ColorPalette.Gray200
-@Stable val Grey300 = ColorPalette.Gray300
-@Stable val Grey400 = ColorPalette.Gray400
-@Stable val Grey500 = ColorPalette.Gray500
-@Stable val Grey600 = ColorPalette.Gray600
-@Stable val Grey700 = ColorPalette.Gray700
-@Stable val Grey800 = ColorPalette.Gray800
-@Stable val Grey900 = ColorPalette.Gray900
+import com.trafi.ui.theme.internal.GrayScale
```
Comment (justasm) [inline android/ui/src/main/java/com/trafi/ui/theme/Color.kt:11]: If we're exposing `GrayScale` to our SDK users, it should be in the `com.trafi.ui.theme` package (not `internal`)

**Review by justasm — APPROVED** [review]
So the main change with this PR is that grayscale is theme-able by SDK users and is dark-theme aware.

Oh, and please ask designers to sync their color palette on the canvas and in the sidebar styles.
Screen Shot 2020-12-15 at 12 36 44

**Review by justasm — APPROVED** [review]

**Comment by mikasdo** [comment]
@eligijusTrafi After some renamings and changes iOS was not updated
We should not merge to main before both platforms are successfuly build ✅

**Pull request merged**